pkg/secrethub/credentials/readers.go

```diff
@@ -14,16 +14,17 @@ type Reader interface {
 // FromFile returns a reader that reads the contents of a file,
 // e.g. a credential or a passphrase.
 func FromFile(path string) Reader {
-	return readerFunc(func() ([]byte, error) {
+	return newReader(path, func() ([]byte, error) {
 		return ioutil.ReadFile(path)
 	})
 }
 
 // FromEnv returns a reader that reads the contents of an
 // environment variable, e.g. a credential or a passphrase.
 func FromEnv(key string) Reader {
-	return readerFunc(func() ([]byte, error) {
-		return []byte(os.Getenv(key)), nil
+	credential := os.Getenv(key)
+	return newReader("$"+key, func() ([]byte, error) {
+		return []byte(credential), nil
 	})
```

Comment:
This technically still breaks backwards compatibility, but only if someone relied on the delayed loading of the credential with this function. I think it's not a big issue, but we can create a separate function to maintain this functionality if needed.

Reply:
I don't see any issues in changing the evaluation order.

However, I also don't see any reason to change it. Couldn't we just keep evaluation as is by coding it like this:

```Golang
func FromEnv(key string) Reader {
	return newReader("$"+key, func() ([]byte, error) {
		return []byte(os.Getenv(key)), nil
	})
}
```
```

Reply:
Yeah you're right. I implemented your suggestion.